loader/proto.go

```diff
@@ -751,6 +723,454 @@ func (b *builder) handlePackage() (string, error) {
 	return w.String(), nil
 }
 
+func isGoType(typ string) bool {
+	switch typ {
+	case "bool",
+		"string",
+		"[]byte",
+		"float64",
+		"float32",
+		"int",
+		"int32",
+		"int64",
+		"uint32",
+		"uint64":
+		return true
+	default:
+		return false
+	}
+}
+
+func (b *builder) fromStructToAnnotation(val interface{}) string {
+	w := &strings.Builder{}
+
+	s := reflect.TypeOf(val)
+	v := reflect.ValueOf(val)
+
+	for i := 0; i < s.NumField(); i++ {
+		name := s.Field(i).Name
+		value := v.FieldByName(name)
+		if !isNilOrEmpty(value.Interface()) {
+			k := value.Kind()
+			switch {
+			case k >= reflect.Int && k <= reflect.Int64,
+				k == reflect.Float32 || k == reflect.Float64,
+				k >= reflect.Uint && k <= reflect.Uint64,
+				k == reflect.String,
+				k == reflect.Bool:
+				// The only way to know if the value
+				// is of type alias is to check its type.
+				if isGoType(value.Type().String()) {
+					b.format(w, 0, nil, "// %s: %v,\n", name, value)
+				} else {
+					pkg := strings.Split(reflect.TypeOf(value.Interface()).String(), ".")[0]
+					b.format(w, 0, nil, "// %s: %s.%v,\n", name, pkg, value)
+				}
+			case k == reflect.Map:
+				mapKey := value.Type().Key()
+				mapValue := strings.Replace(value.Type().Elem().String(), "*", "", -1)
+				b.format(w, 0, nil, "// %s: map[%s]%s{\n", name, mapKey, mapValue)
+				for _, key := range value.MapKeys() {
+					c := value.MapIndex(key)
+					switch reflect.TypeOf(c.Interface()).Kind() {
+					case reflect.Ptr:
+						t := reflect.Indirect(c).Interface()
+						b.format(w, 0, nil, "// %s: %s{\n", key, reflect.TypeOf(t))
+						b.format(w, 0, nil, b.fromStructToAnnotation(t))
+						b.format(w, 0, nil, "// },\n")
+					default:
+						b.format(w, 0, nil, "// %s: %s,\n", key, c)
+					}
+				}
+				b.format(w, 0, nil, "// },\n")
+			case k == reflect.Ptr:
+				t := reflect.Indirect(value).Interface()
+				b.format(w, 0, nil, "// %s: %s{\n", name, reflect.TypeOf(t))
+				w.WriteString(b.fromStructToAnnotation(t))
+				b.format(w, 0, nil, "// },\n")
+			case k == reflect.Slice:
+				b.format(w, 0, nil, "// %s: %s{\n", name, reflect.TypeOf(value.Interface()))
+				for i := 0; i < value.Len(); i++ {
+					val := value.Index(i).Interface()
+					switch reflect.TypeOf(val).Kind() {
+					case reflect.Ptr:
+						// TODO (ced)
+					default:
+						if isGoType(reflect.TypeOf(val).String()) {
```

Comment:
This seems a bit weird. When would a `TypeOf` not be a go type?

Also, whatever you want to know about the type, you can do so without string handling. For example, to know if a type is named versus a simple Go type, just use the `Name() string` method.

Reply:
When you have an enum, the `Kind()` will be the underlying type of the alias and the  `TypeOf()` is the alias. In the gunk annotation I want to display the enum name and not its underlying value: 
```
Type: openapiv2.TYPE_API_KEY ,
```
for instance and not 
```
Type: 1,
```

Reply:
If you want the name of a named type, you can use the Name method - otherwise Kind can tell you what kind the type is. String is too sloppy for this stuff, IMO. But we can clean this up after this PR if you're not sure what I mean.